File: ProjectStart/ProjectStart.WebApp/ClientApp/Shared/AppModels/PageDataEntity.ts
import { INodeEntity, NodeEntity} from 'shared/AppModels/NodeEntity';
/**
 * Interface for: ProjectStart.Entity.Cms.PageDataEntity
 */
export interface IPageDataEntity extends INodeEntity<PageDataEntity> {
    title: string;
    description: string;
    keywords: string;
    url: string;
    mode?: ModeEnum;
}
/**
 * Base view model for ProjectStart.Entity.Cms.PageDataEntity
 */
export class PageDataEntity extends NodeEntity<PageDataEntity> implements IPageDataEntity {
    title: string;
    description: string;
    keywords: string;
    url: string;
    mode?: ModeEnum;
}

File: ProjectStart/ProjectStart.Entity/Cms/PageDataEntity.cs
using ProjectStart.Entity.Common;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace ProjectStart.Entity.Cms
{
    [Table("PageData")]
    public class PageDataEntity : NodeEntity<PageDataEntity>, IPageContent
    {
        [Required]
        public string Title { get; set; }
        [Required]
        public string Description { get; set; }
        [Required]
        public string Keywords { get; set; }
        public string Url { get; set; }
    }
}
